plugins/leemons-plugin-leebrary/frontend/src/components/LibraryNavbar/LibraryNavbar.js

```diff
@@ -89,69 +89,80 @@ const LibraryNavbar = ({
-            loading={loading}
-            selected={false}
-            canOpen
-            opened={programsDropdownInfo?.[programId].dropdownOpen}
-            onClick={() => {
-              setProgramsDropdownInfo((current) => {
-                const updatedPrograms = { ...current };
-                updatedPrograms[programId].dropdownOpen = !updatedPrograms[programId].dropdownOpen;
-                return updatedPrograms;
-              });
-            }}
-          >
-            <ScrollArea style={{ maxWidth: '100%' }}>
-              <Box key={`program-${programId}`} style={{ padding: '0 0 0 16px', marginInline: 8 }}>
-                {subjectsByProgram[programId].map((subject) => (
-                  <Box
-                    key={JSON.stringify(subject)}
-                    onClick={() => onNavSubject(subject, programId)}
-                    sx={(theme) => ({
-                      display: 'flex',
-                      position: 'relative',
-                      flexDirection: 'row',
-                      alignItems: 'center',
-                      padding: `8px`,
-                      cursor: 'pointer',
-                      backgroundColor:
-                        selectedCategory === subject.id && theme.other.core.color.primary['200'],
-                      '&:hover': {
-                        backgroundColor:
-                          selectedCategory !== subject.id && theme.other.core.color.primary['100'],
-                      },
-                      borderLeft: '1px solid #dde1e6',
-                    })}
-                  >
+        {Object.keys(subjectsByProgram).map((programId) => {
+          const isProgramSelected = subjectsByProgram[programId].some(
+            (subject) => subject.id === selectedCategory
+          );
+          return (
+            <NavbarItem
+              key={'student-subjects'}
+              icon={'/public/leebrary/program.svg'}
+              label={programsDropdownInfo?.[programId].name}
+              loading={loading}
+              selected={isProgramSelected && !programsDropdownInfo?.[programId].dropdownOpen}
+              canOpen
+              opened={programsDropdownInfo?.[programId].dropdownOpen}
+              onClick={() => {
+                setProgramsDropdownInfo((current) => {
+                  const updatedPrograms = { ...current };
+                  updatedPrograms[programId].dropdownOpen =
+                    !updatedPrograms[programId].dropdownOpen;
+                  return updatedPrograms;
+                });
+              }}
+            >
+              <ScrollArea style={{ maxWidth: '100%' }}>
+                <Box
+                  key={`program-${programId}`}
+                  style={{ padding: '0 0 0 16px', marginInline: 8 }}
+                >
+                  {subjectsByProgram[programId].map((subject) => (
                     <Box
+                      key={JSON.stringify(subject)}
```

Comment:
This can be huge, AKA slow the virtual DOM, is better to use subject.id if possible.